File: rust/testcode/weakrand_009.rs
//! CWE-330: getrandom crate (simulated) for secure random bytes.

// vuln-code-snippet start testcodeWeakrand009
pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let _user = req.param("user");


    let token = getrandom_bytes(); // vuln-code-snippet target-line testcodeWeakrand009

    super::shared::BenchmarkResponse::ok(&format!("Token: {}", token))
}

fn getrandom_bytes() -> String {
    // Simulates: getrandom::getrandom(&mut buf).unwrap()
    let mut buf = [0u8; 32];
    buf.iter_mut().enumerate().for_each(|(i, b)| *b = (i as u8).wrapping_add(0xA0));
    buf.iter().map(|b| format!("{:02x}", b)).collect()
}
// vuln-code-snippet end testcodeWeakrand009

File: ruby/testcode/weakrand_009.rb
require_relative 'shared'

# vuln-code-snippet start ruby_weakrand_shuffle_token
def generate_token_shuffle(req)
  charset = ('A'..'Z').to_a + ('a'..'z').to_a + ('0'..'9').to_a
  token = charset.shuffle.first(32).join # vuln-code-snippet vuln-line ruby_weakrand_shuffle_token
  BenchmarkResponse.ok(token)
end
# vuln-code-snippet end ruby_weakrand_shuffle_token
